planner/core/cacheable_checker.go

```diff
@@ -105,10 +110,27 @@ func (checker *cacheableChecker) Enter(in ast.Node) (out ast.Node, skipChildren
 			checker.cacheable = false
 			return in, true
 		}
+	case *ast.TableName:
+		if checker.isPartitionTable(node) {
+			checker.cacheable = false
+			return in, true
+		}
 	}
 	return in, false
 }
 
+func (checker *cacheableChecker) isPartitionTable(tn *ast.TableName) bool {
+	tb, err := checker.schema.TableByName(tn.Schema, tn.Name)
```

Comment:
This change may have some performance impact.

Reply:
> Why would this PR incur significant performance regression? it just adds one more check for TableName.

Because of the check here. @eurekaka

Reply:
Are these 2 map lookups so expensive?

Reply:
schema.TableByName() is `O(N)` operation.
And the `cacheableChecker` itself need to handle each tableName in the AST.